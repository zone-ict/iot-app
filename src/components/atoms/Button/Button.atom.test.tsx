// We're unit testing so this is fine.
/* eslint-disable @typescript-eslint/no-unsafe-call */

import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button.atom';

describe('Button regular component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button>Button Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly when there's no child component", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with text child and type', () => {
    const tree = renderer.create(<Button type="submit">Button Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when component is disabled', () => {
    const tree = renderer.create(<Button disabled>Button Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to fire onClick event', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Button Text</Button>);
    fireEvent.click(screen.getByText('Button Text'));
    expect(onClick).toHaveBeenCalled();
  });

  it('should render correctly with small prop', () => {
    const tree = renderer.create(<Button small>Button Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with flat prop', () => {
    const tree = renderer.create(<Button flat>Button Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Button Outlined component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button.Outlined>Button Text</Button.Outlined>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly when there's no child component", () => {
    const tree = renderer.create(<Button.Outlined />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with text child and type', () => {
    const tree = renderer
      .create(<Button.Outlined type="submit">Button Text</Button.Outlined>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when component is disabled', () => {
    const tree = renderer.create(<Button.Outlined disabled>Button Text</Button.Outlined>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to fire onClick event', () => {
    const onClick = jest.fn();
    const tree = renderer.create(<Button.Outlined onClick={onClick} />);
    tree.root.props.onClick();
    expect(onClick).toHaveBeenCalled();
  });

  it('should render correctly with small prop', () => {
    const tree = renderer.create(<Button.Outlined small>Button Text</Button.Outlined>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with flat prop', () => {
    const tree = renderer.create(<Button.Outlined flat>Button Text</Button.Outlined>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Button Text component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button.Outlined>Button Text</Button.Outlined>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly when there's no child component", () => {
    const tree = renderer.create(<Button.Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with text child and type', () => {
    const tree = renderer.create(<Button.Text type="submit">Button Text</Button.Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when component is disabled', () => {
    const tree = renderer.create(<Button.Text disabled>Button Text</Button.Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to fire onClick event', () => {
    const onClick = jest.fn();
    const tree = renderer.create(<Button.Text onClick={onClick} />);
    tree.root.props.onClick();
    expect(onClick).toHaveBeenCalled();
  });

  it('should render correctly with small prop', () => {
    const tree = renderer.create(<Button.Text small>Button Text</Button.Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
