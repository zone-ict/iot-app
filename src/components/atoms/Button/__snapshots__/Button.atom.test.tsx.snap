// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Outlined component should render correctly 1`] = `
<button
  className="css-1lzjpho"
>
  Button Text
</button>
`;

exports[`Button Outlined component should render correctly when component is disabled 1`] = `
<button
  className="css-1lhoyk"
  disabled={true}
>
  Button Text
</button>
`;

exports[`Button Outlined component should render correctly when there's no child component 1`] = `
<button
  className="css-1lzjpho"
/>
`;

exports[`Button Outlined component should render correctly with flat prop 1`] = `
<button
  className="css-pwy6z3"
>
  Button Text
</button>
`;

exports[`Button Outlined component should render correctly with small prop 1`] = `
<button
  className="css-ndby1y"
>
  Button Text
</button>
`;

exports[`Button Outlined component should render correctly with text child and type 1`] = `
<button
  className="css-1lzjpho"
  type="submit"
>
  Button Text
</button>
`;

exports[`Button Text component should render correctly 1`] = `
<button
  className="css-1lzjpho"
>
  Button Text
</button>
`;

exports[`Button Text component should render correctly when component is disabled 1`] = `
<button
  className="css-emjo5x"
  disabled={true}
>
  Button Text
</button>
`;

exports[`Button Text component should render correctly when there's no child component 1`] = `
<button
  className="css-1oskd8"
/>
`;

exports[`Button Text component should render correctly with small prop 1`] = `
<button
  className="css-d5oupz"
>
  Button Text
</button>
`;

exports[`Button Text component should render correctly with text child and type 1`] = `
<button
  className="css-1oskd8"
  type="submit"
>
  Button Text
</button>
`;

exports[`Button regular component should render correctly 1`] = `
<button
  className="css-qrgu0p"
>
  Button Text
</button>
`;

exports[`Button regular component should render correctly when component is disabled 1`] = `
<button
  className="css-zs6jsq"
  disabled={true}
>
  Button Text
</button>
`;

exports[`Button regular component should render correctly when there's no child component 1`] = `
<button
  className="css-qrgu0p"
/>
`;

exports[`Button regular component should render correctly with flat prop 1`] = `
<button
  className="css-1xavz1o"
>
  Button Text
</button>
`;

exports[`Button regular component should render correctly with small prop 1`] = `
<button
  className="css-1huf3kw"
>
  Button Text
</button>
`;

exports[`Button regular component should render correctly with text child and type 1`] = `
<button
  className="css-qrgu0p"
  type="submit"
>
  Button Text
</button>
`;
